src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -3598,17 +3611,24 @@ private void onApplicationData(
                         doNetworkAbort(traceId, authorization);
                     }
 
-                    nextOutgoingId++;
-                    outgoingWindow--;
-
-                    if ((flags & FLAG_INIT) == FLAG_INIT)
+                    if ((flags & FLAG_INIT_INCOMPLETE) != FLAG_INIT_INCOMPLETE)
                     {
-                        deliveryId++;
-                        onApplicationDataInit(traceId, reserved, authorization, flags, extension, payload);
+                        nextOutgoingId++;
+                        outgoingWindow--;
+
+                        if ((flags & FLAG_INIT) == FLAG_INIT)
+                        {
+                            deliveryId++;
+                            onApplicationDataInit(traceId, reserved, authorization, flags, extension, payload);
+                        }
+                        else if (deliveryId != abortedDeliveryId)
+                        {
+                            onApplicationDataContOrFin(traceId, reserved, authorization, flags, payload);
+                        }
                     }
-                    else if (deliveryId != abortedDeliveryId)
+                    else
                     {
-                        onApplicationDataContOrFin(traceId, reserved, authorization, flags, payload);
+                        flushReplySharedBudget(traceId);
```

Comment:
Suggest inverting this `if-else` such that the `if` condition handles `incomplete` as the special case, falling through to normal case in `else`.

It has the same efficiency at runtime, but makes it a bit easier to read the code.